src/Workspaces/Remote/ServiceHub/Services/ServiceHubDocumentTrackingService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    [ExportWorkspaceService(typeof(IDocumentTrackingService), ServiceLayer.Host)]
+    [Shared]
+    internal sealed class ServiceHubDocumentTrackingService : IDocumentTrackingService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ServiceHubDocumentTrackingService()
+        {
+        }
+
+        public bool SupportsDocumentTracking => false;
+
+        public event EventHandler<DocumentId?> ActiveDocumentChanged { add { } remove { } }
+        public event EventHandler<EventArgs> NonRoslynBufferTextChanged { add { } remove { } }
+
+        public ImmutableArray<DocumentId> GetVisibleDocuments()
+        {
+            Fail("Code should not be attempting to obtain visible documents from a stateless remote invocation.");
+            return ImmutableArray<DocumentId>.Empty;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
why?

Reply:
Accidental. I needed to find the name `WatsonReporter.ReportNonFatal`. I wasn't sure what the name was, but I knew it used `MethodImplOptions.NoInlining` as part of setting the stack trace. I just forgot to remove this line after finding references.

Reply:
:D